orchestrator/careplancontributor/service.go

```diff
@@ -182,15 +188,21 @@ func (s Service) handleProxyToFHIR(writer http.ResponseWriter, request *http.Req
 	for _, careTeam := range careTeams {
 		for _, participant := range careTeam.Participant {
 			for _, identifier := range principal.Organization.Identifier {
-				if coolfhir.IdentifierEquals(participant.Member.Identifier, &identifier) {
+				if coolfhir.IdentifierEquals(participant.OnBehalfOf.Identifier, &identifier) {
 					// Member must have start date, this date must be in the past, and if there is an end date then it must be in the future
-					err = coolfhir.ValidateCareTeamParticipantPeriod(participant)
+					err = coolfhir.ValidateCareTeamParticipantPeriod(participant, time.Now())
 					if err != nil {
 						return err
 					}
 					isValidRequester = true
 					break
 				}
+				if isValidRequester {
+					break
+				}
+			}
+			if isValidRequester {
+				break
 			}
 		}
 	}
```

Comment:
the multiple if-breaks make this hard to grasp, my solution would be a separate func where you can just `return` from, or add a GOTO-ish label to the outer for-loop (and use `break outer/(or something`)

Reply:
Yeah I am not happy with this either, I will clean it up. Thanks